src/iota/essence.c

```diff
@@ -318,8 +302,8 @@ static uint8_t validate_inputs_lexical_order(const UTXO_INPUT *inputs,
 
     for (uint32_t i = 0; i < inputs_count - 1; i++) {
         // Inputs must be in lexicographical order of their serialized form.
-        if (memcmp_bytewise((uint8_t *)&inputs[i], (uint8_t *)&inputs[i + 1],
-                            sizeof(UTXO_INPUT)) != -1) {
+        if (memcmp((uint8_t *)&inputs[i], (uint8_t *)&inputs[i + 1],
```

Comment:
according to `memcmp`:
The `memcmp()` function returns an integer less than, equal to, or greater  than  zero if  the first n bytes of s1 is found, respectively, to be less than, to match, or be greater than the first n bytes of s2.